src/cluster/pager.rs

```diff
@@ -39,6 +39,50 @@ impl<
     }
   }
 
+
+  /// Java native driver also uses this technique to
+  /// perform stateless pagination
+  /// (https://docs.datastax.com/en/developer/java-driver/3.2/manual/paging/)
+  pub fn query_with_pager_state<Q>(&'a mut self, query: Q, state: Option<String>) -> Result<QueryPager<'a, Q, SessionPager<'a, M, S, T>>, error::Error>
+    where
+      Q: ToString,
+  {
+    let tmp_state: Option<Result<Option<CBytes>, &str>> = state
```

Comment:
I'm not quite getting why we need to transform `CBytes` to a string and then, here, we need to decode it back to `CBytes`. This extra logic requires an extra error handling here.

For me it sounds reasonable to use such `PagerState` struct that it requires near to zero transformations.

Reply:
I have removed the encoding and decoding and transfer that responsibility for the library user. 

`CBytes` into `string` is a way to send to the HTTP client the pager state in representable characters.

An alternative was to `CBytes` into `string` represented in base64, so is better for the lib user to pick what is best for moving around the state.